crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -90,6 +91,17 @@ pub(super) trait RoomDataProvider:
     async fn load_fully_read_marker(&self) -> Option<OwnedEventId>;
 
     async fn push_rules_and_context(&self) -> Option<(Ruleset, PushConditionRoomCtx)>;
+
+    /// Send an event to the fake room.
+    async fn send(&self, content: AnyMessageLikeEventContent) -> Result<(), super::Error>;
+
+    /// Redact an event in the fake room.
+    async fn redact(
+        &self,
+        event_id: &EventId,
+        reason: Option<&str>,
+        transaction_id: Option<OwnedTransactionId>,
+    ) -> Result<(), super::Error>;
```

Comment:
Why “fake room”?

Could this change be part of a single commit?

Reply:
Oops, `fake room` is from a previous version of the code where this was on the `TestTimeline` impl. Will see how hard it is to isolate it into its own commit first :pray: